cpp/src/arrow/engine/substrait/options.h

```diff
@@ -76,13 +88,22 @@ class ARROW_ENGINE_EXPORT ExtensionProvider {
  public:
   static std::shared_ptr<ExtensionProvider> kDefaultExtensionProvider;
   virtual ~ExtensionProvider() = default;
-  virtual Result<RelationInfo> MakeRel(const std::vector<DeclarationInfo>& inputs,
+  virtual Result<RelationInfo> MakeRel(const ConversionOptions& conv_opts,
+                                       const std::vector<DeclarationInfo>& inputs,
                                        const ExtensionDetails& ext_details,
                                        const ExtensionSet& ext_set) = 0;
 };
 
 ARROW_ENGINE_EXPORT std::shared_ptr<ExtensionProvider> default_extension_provider();
 
+struct ARROW_ENGINE_EXPORT NamedTapNodeOptions : public compute::ExecNodeOptions {
```

Comment:
We can pass the schema and name directly instead of using this class.  If we still want to wrap the two of them in some kind of object (to reduce the # of args passed into the mapper?) then we should include `kind`, do not extend `ExecNodeOptions`, and call it something like NamedTap or NamedTapOptions.

In other words, we shouldn't try to form the node options from the protobuf.  That will be the mapper's job.  We just need to pass whatever was in the protobuf along to the mapper and let the mapper figure out what node options are appropriate.

Reply:
Done.